python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -75,6 +76,7 @@ def from_data_product_metadata_dict(metadata: dict, version, owner_id: str):
             status=DataProductStatus[metadata["status"]],
             retention_period_in_days=metadata["retentionPeriod"],
             domain=metadata["domain"],
```

Comment:
Need to ensure domain isn't an empty string.

Maybe we can make use of [pydantic (or pydantic style) validators](https://docs.pydantic.dev/latest/concepts/validators/), e.g. `min_length=1` for string values

Reply:
Domain is mandatory so should never be an provided as an empty string but this should be validated as you say. Perhaps one for another ticket.